src/pathfind/teleport.hpp

```diff
@@ -120,13 +122,13 @@ class teleport_map {
 	 * @param loc			the map location for which we want to know the adjacent hexes
 	 * @todo what does this function actually have to do with adjacent hexes?
 	 */
-	std::set<map_location> get_adjacents(map_location loc) const;
+	const std::unordered_set<map_location, map_location_hasher>& get_adjacents(map_location loc) const;
```

Comment:
I'm noticing the `@todo` here and offering an answer, so this comment isn't really related to your PR. The "adjacency" considered by this function is a different form of "adjacency" than that considered by the function to get all hexes adjacent to a given hex – it refers to "adjacency on the graph", that is, there is an edge connecting the two nodes (or, to put it another way, there is a connection allowing you to traverse directly between from this hex to the other one).